### PR TITLE
avm2: Implement a few XML/XMLlist methods

### PR DESCRIPTION
**Pull request by Dinnerbone**
Nothing too exciting, just:

- `processingInstructions(name)` + make processingInstructions actually parse/stringify correctly
- `comments()`
- `valueOf()`
- `toJSON()`

### CODE REVIEW

File: core/src/avm2/e4x.rs [hunk 1]
```diff
@@ -711,8 +742,13 @@ fn to_xml_string_inner<'gc>(xml: E4XOrXml<'gc>, buf: &mut WString) -> Result<(),
             buf.push_str(&escape_element_value(*text));
```
Comment (Lord-McSweeney) [inline core/src/avm2/e4x.rs:742]: `to_xml_string_inner` will never return an Error now, so I think its return type can be removed.
This also means `to_xml_string` (previously returned an `Result<AvmString, Error>`) can now return an `AmString<'gc>`.

File: core/src/avm2/globals/XMLList.as [hunk 2]
```diff
@@ -21,11 +21,24 @@ package {
...
         AS3 native function toString():String;
+        AS3 native function comments():XMLList;
+        AS3 native function processingInstructions(name:String = "*"):XMLList;
 
         // The following native methods are not declared in the documentation,
         // but still exist
         AS3 native function name(): Object;
 
+        AS3 function toJSON(k:String) : *
+        {
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/XMLList.as:32]: nit: The rest of the file puts the bracket on the same line as the function definition, like `AS3 function toJSON(k:String):* {`

File: core/src/avm2/e4x.rs (hunk 1 above)
Comment (Dinnerbone) [reply core/src/avm2/e4x.rs:742]: Good catch!

File: core/src/avm2/globals/XMLList.as (hunk 2 above)
Comment (Dinnerbone) [reply core/src/avm2/globals/XMLList.as]: Fixed

File: core/src/avm2/e4x.rs [hunk 3]
```diff
@@ -774,7 +808,7 @@ pub fn to_xml_string<'gc>(
     activation: &mut Activation<'_, 'gc>,
 ) -> Result<AvmString<'gc>, Error<'gc>> {
```
Comment (Lord-McSweeney) [inline core/src/avm2/e4x.rs:809]: This can return an `AvmString<'gc>` now since it will always return `Ok`

Comment (Dinnerbone) [reply core/src/avm2/e4x.rs]: Fixed (and a few other methods)

**Review by Lord-McSweeney — APPROVED** [review]

**Review by adrian17 — APPROVED** [review]

**Pull request merged**